File: Easy/Filter/Others/1 Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        num_dict = {}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_dict:
                return ([num_dict[complement], i])
            num_dict[num] = i


S = Solution()
print(S.twoSum([2, 7, 11, 15], 9))

File: PHP/Easy/Filters/Hash Table/1__Two_Sum.php
<?php
//https://leetcode.com/problems/two-sum/description/

function twoSum($nums, $target)
{
    $num_dict = [];
    foreach ($nums as $i => $num) {
        $complement = $target - $num;
        if (array_key_exists($complement, $num_dict)) {
            return [$num_dict[$complement], $i];
        }
        $num_dict[$num] = $i;
    }
    return [];
}

$nums = [2, 7, 11, 15];
$target = 9;
$result = twoSum($nums, $target);
print_r($result);